src/nvs.rs

```diff
@@ -810,28 +737,186 @@ impl RawHandle for EspNvs<NvsEncrypted> {
+            return match data_type {
+                NvsDataType::Blob => self.0.blob_len(name),
+                NvsDataType::U64 => {
+                    let value = self.0.get_u64(name)?;
+                    if value.is_none() {
+                        return Ok(None);
+                    }
+                    let value = value.unwrap();
+                    // u64 value was found, decode it by getting the first byte as length
+                    let len: u8 = (value & 0xff) as u8;
+
+                    Ok(Some(len as _))
+                }
+                _ => Ok(None), // Other types are not supported in this storage
+            };
+        }
+        Ok(None)
+    }
+
+    pub fn get_raw<'a>(&self, name: &str, buf: &'a mut [u8]) -> Result<Option<&'a [u8]>, EspError> {
+        if let Some(data_type) = self.0.find_key(name)? {
+            return match data_type {
+                NvsDataType::Blob => self.0.get_blob(name, buf),
+                NvsDataType::U64 => {
+                    if let Some(mut u64value) = self.0.get_u64(name)? {
+                        // The first byte is the length
+                        let len = (u64value & 0xff) as u8;
+
+                        if buf.len() < len as _ {
+                            return Err(EspError::from_infallible::<ESP_ERR_NVS_INVALID_LENGTH>());
+                        }
+
+                        u64value >>= 8;
+                        let array: [u8; 7] = [
+                            (u64value & 0xff) as u8,
+                            ((u64value >> 8) & 0xff) as u8,
+                            ((u64value >> 16) & 0xff) as u8,
+                            ((u64value >> 24) & 0xff) as u8,
+                            ((u64value >> 32) & 0xff) as u8,
+                            ((u64value >> 40) & 0xff) as u8,
+                            ((u64value >> 48) & 0xff) as u8,
+                        ];
+
+                        buf[..len as usize].copy_from_slice(&array[..len as usize]);
+
+                        Ok(Some(&buf[..len as usize]))
+                    } else {
+                        Ok(None)
+                    }
+                }
+                _ => Ok(None), // Other types are not supported in this storage
+            };
+        }
+        Ok(None)
+    }
+
+    pub fn set_raw(&self, name: &str, buf: &[u8]) -> Result<bool, EspError> {
+        // start by just clearing this key, ignoring the result since it may not exist
+        _ = self.0.remove(name);
+
+        if buf.len() < 8 {
+            // If the buffer is less than 8 bytes, store it as a u64
+            let mut u64value: u_int64_t = 0;
+
+            // Set the first byte as the length
+            u64value |= buf.len() as u_int64_t;
+
+            // Fill the rest of the u64 with the buffer values
+            for v in buf.iter().rev() {
+                u64value <<= 8;
+                u64value |= *v as u_int64_t;
+            }
+
+            u64value <<= 8;
```

Comment:
Something is still not quite right here... was it _really_ like this? As in... writing the len twice?

Reply:
No! the original version write the len after the for loop, I will keep the original one, by  removing the len write before the loop.